app/src/main/java/com/example/mystore/ui/home/filters/ExpandableList.kt

```diff
@@ -47,13 +47,11 @@ fun ExpandableList(
     onFiltersChanged: (CategoryDetailsModel) -> Unit
 ) {
     var expanded by remember { mutableStateOf(false) }
-    val selectedCategoriesString: String = stringResource(id = R.string.all_categories_filter)
-    val filteredCategories2: List<CategoryDetailsModel> =
-        data.filter { selectedCategoriesString.contains(it.name) }
-    if (filteredCategories2.isNotEmpty() && filteredCategories2.count() != data.count()) {
-        filteredCategories2.joinToString { it.name }
-    } 
-
+    var selectedCategoriesString = stringResource(id = R.string.all_categories_filter)
```

Comment:
Make it into a getter, along with the `if` clause logic.

Reply:
In the if clause i have property  **filteredCategories** which is not initialized yet. How can i use it in the `if` clause?

Reply:
Doesn't something like this work?

```
val filteredCategories get() = data.filter { selectedCategories.contains(it.name) }

val selectedCategoriesString get() {
  return if (filteredCategories.isNotEmpty() && filteredCategories.count() != data.count()) {
   filteredCategories.joinToString { it.name }
  } else {
   stringResource(id = R.string.all_categories_filter)
}
```

Reply:
It is not working, maybe because it is in composable

Reply:
Oh right...

Then you can should be able to do the same thing with function variables:


```
val filteredCategories: () -> List<String> = {
  data.filter { selectedCategories.contains(it.name) }
}

val selectedCategoriesString: () -> String = {
 if (filteredCategories.isNotEmpty() && filteredCategories.count() != data.count()) {
   filteredCategories.joinToString { it.name }
  } else {
   stringResource(id = R.string.all_categories_filter)
}
```